tasks/core/config.yml

```diff
@@ -37,18 +43,13 @@
       register: result
       changed_when: result is success
       notify: nextcloud update htaccess
+      failed_when: (not (result.stdout | search('successfully imported'))
+        or result.rc != 0)
 
     - meta: flush_handlers
 
     - name: Delete config file
       file:
         path: /tmp/nextcloud_config_53ab3c4
         state: absent
-  when: _nextcloud_prefs_need_updating | bool
-  vars:
-    # Check if nextcloud_config is not a subset of the old preferences
-    _nextcloud_prefs_need_updating: >-
-      {{
-        (nextcloud_config|dict_flatten).items()
-        > (_nextcloud_old_preferences|dict_flatten).items()
-      }}
+  when: not (_nextcloud_updated_preferences == _nextcloud_old_preferences)
```

Comment:
Why not replace `==` with `!=` and remove `not`?

Reply:
Ah, sorry, leftover from the logic before I cleaned up. Will fix.